arch/x86/entry.S

```diff
@@ -164,6 +186,28 @@ ENTRY(terminate_user_task)
     ret
 END_FUNC(terminate_user_task)
 
+.align PAGE_SIZE
+ENTRY(syscall_handler_entry)
+    syscall_from_usermode
+
+    cmp $SYSCALL_EXIT, %_ASM_AX
+    jz syscall_exit
+
+    push %_ASM_CX
+    push %r11
+
+    mov %_ASM_DI, %_ASM_CX
+    mov %_ASM_AX, %_ASM_DI
+    call syscall_handler
+
+    pop %r11
+    pop %_ASM_CX
+
+    syscall_to_usermode
+
+    SYSRET
```

Comment:
I don't know what guarantees KTF wants to provide for the syscall ABI. But right now it clobbers all the caller-saved registers (RAX (intentional, as return value register), RDX, RSI, R8-R10). Or, to read it differently, it _leaks_ those registers to userland.

We might want to do a follow up patch that sanitizes the callee-clobbered registers to:
1. avoid speculation based on user provided register values and
2. prevent leaking kernel data to userland / clobber user register state.

Reply:
Modulo the callee saved registers handled in kernel instead of user mode, I am not sure we care much about the security aspects. After all, we always run trusted code that does malicious things in KTF :-)

Reply:
I was just concerned about getting test flawed results by seeing leaked kernel registers.